state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -217,6 +213,51 @@ pub enum PalletConfigUpdate<T: Config<I>, I: 'static = ()> {
 	SetMaxSwapRetryDurationBlocks { blocks: BlockNumber },
 }
 
+macro_rules! append_chain_to_name {
+	($name:ident) => {
+		match T::TargetChain::NAME {
+			"Ethereum" => concat!(stringify!($name), "Ethereum"),
+			"Polkadot" => concat!(stringify!($name), "Polkadot"),
+			"Bitcoin" => concat!(stringify!($name), "Bitcoin"),
+			"Arbitrum" => concat!(stringify!($name), "Arbitrum"),
+			"Solana" => concat!(stringify!($name), "Solana"),
+			_ => concat!(stringify!($name), "Other"),
+		}
+	};
+}
+
+impl<T, I> TypeInfo for PalletConfigUpdate<T, I>
+where
+	T: Config<I>,
+	I: 'static,
+{
+	type Identity = Self;
+	fn type_info() -> Type {
+		Type::builder()
+			.path(Path::new(append_chain_to_name!(PalletConfigUpdate), module_path!()))
+			.variant(
+				Variants::new()
+					.variant("ChannelOpeningFee", |v| {
+						v.index(0)
+							.fields(Fields::named().field(|f| f.ty::<T::Amount>().name("fee")))
+					})
+					.variant(append_chain_to_name!(SetMinimumDeposit), |v| {
```

Comment:
why is this one different? why do we append the chain name here but not the others?

Reply:
this is the only one that has a different type across palettes. I tried to keep the renaming to the absolute minimum that still fixes the bug. This is the result of a lot of trial and error.